homeassistant/components/v2c/diagnostics.py

```diff
@@ -0,0 +1,35 @@
+"""Diagnostics support for V2C."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any
+
+from homeassistant.components.diagnostics import async_redact_data
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST
+from homeassistant.core import HomeAssistant
+
+from .const import DOMAIN
+from .coordinator import V2CUpdateCoordinator
+
+TO_REDACT = {CONF_HOST, "title"}
+
+
+async def async_get_config_entry_diagnostics(
+    hass: HomeAssistant, entry: ConfigEntry
+) -> dict[str, Any]:
+    """Return diagnostics for a config entry."""
+    coordinator: V2CUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
```

Comment:
Side note: I think this would benefit from migration to `entry.runtime_data`

Reply:
@epenet https://github.com/home-assistant/core/pull/119165